src/apps/external_executor/external_executor.cpp

```diff
@@ -692,7 +691,12 @@ namespace externalexecutor
           out_stream->stream_msg(result);
         }
 
-        return ccf::grpc::make_success();
+        ctx.rpc_ctx->set_response_trailer(
+          ccf::grpc::make_status_trailer(GRPC_STATUS_OK));
+        ctx.rpc_ctx->set_response_trailer(
+          ccf::grpc::make_message_trailer(grpc_status_str(GRPC_STATUS_OK)));
+
+        return ccf::grpc::make_pending();
```

Comment:
Why do we need this here as the endpoint does not detach the stream?

Reply:
We can't use `make_success`, as it needs a type, and we want to avoid templating this on a type. To be passed to `grpc_command_unary_stream_adapter`, we need to return something. We could distinguish "I've detached => pending" from "I've finished => send response now", but I don't think there's much point - the behaviour is the same (something has set the status trailers, send them when the stream closes).

If we think this instance is too weird, we could make it a pure `grpc_adapter` rather than a `unary_stream_adapter` - it's produced its entire body inline, so can return that as a vector (even if gRPC wants to call that a "stream"), so we don't need to include support for detaching?